### PR TITLE
Fix the CI build: Rubocop fixes

### PR DESCRIPTION
**Pull request by orien**
### Context

The main CI build is failing in response to new checks introduced in the latest versions of Rubocop.

### Change

- Resolve issues raised by Rubocop
- Fix a test that was failing on Ruby 3.0
- Run the test suite on Ruby 3.1 during the CI build
- Simplify the CI setup

I recommend running through the changes commit-by-commit for context.

/cc @johnsyweb @joesustaric @domon-envato

### CODE REVIEW

File: unwrappr.gemspec [hunk 1]
```diff
@@ -52,6 +52,7 @@ Gem::Specification.new do |spec| # rubocop:disable Metrics/BlockLength:
   spec.add_development_dependency 'rubocop', '>= 0.49.0'
 
   spec.metadata = {
+    'rubygems_mfa_required' => 'true',
```
Comment (johnsyweb) [inline unwrappr.gemspec:55]: Maybe continue to order attributes alphabetically?

File: lib/unwrappr/octokit.rb [hunk 2]
```diff
@@ -3,6 +3,19 @@
 # Wrapper around octokit
 module Octokit
   def self.client
-    @client ||= Client.new(access_token: ENV['GITHUB_TOKEN'])
+    @client ||= Client.new(access_token: access_token_from_environment)
+  end
+
+  def self.access_token_from_environment
```
Comment (johnsyweb) [inline lib/unwrappr/octokit.rb:9]: Nice extraction!

**Review by johnsyweb — APPROVED** [review]
Thanks so much for picking this up!

File: unwrappr.gemspec (hunk 1 above)
Comment (orien) [reply unwrappr.gemspec]: 👍 Updated in [afb566c](https://github.com/envato/unwrappr/pull/92/commits/afb566c298752f66d5a7d1d7d80a69fd7d5c5fdb)

**Pull request merged**